engine/healthcheck.go

```diff
@@ -243,10 +243,10 @@ func (h *healthcheckManager) newConfig(id healthcheck.Id, key CheckKey, hc *conf
 	port := int(hc.Port)
 	mark := 0
 
-	// For DSR we use the VIP address as the target and specify a mark for
-	// the backend.
+	// For DSR or TUN we use the VIP address as the target and specify a
+	// mark for the backend.
 	ip := host
-	if key.HealthcheckMode == seesaw.HCModeDSR {
+	if key.HealthcheckMode == seesaw.HCModeDSR || key.HealthcheckMode == seesaw.HCModeTUN {
 		ip = key.VserverIP.IP()
 		mark = int(h.markBackend(key.BackendIP))
```

Comment:
Inside markBackend, it creates a ipvs service in gatewaying mode. so it's routing packets in that mode instead of the tunnel. This is requires that the lb node IP has L2 connectivity with the backends. But the benefits of tunneling mode is that LB doesn't have to be in the same L2 subnet as the backends. This doens't seem right to me. Please correct if I'm wrong.

Reply:
You're right. The ipvs service created within `markBackend` also needs to be updated to use `ipvs.DFForwardTunnel`. Missed on that part.

Since the entire (un)markBackend, or more broadly vserver with `use_fwm` are largely not yet covered in test case, let me see if I can refactor `markBackend` a little bit to make it more testable while addressing this tunnel mode healthcheck case.